File: app-code/java/spring-boot-postgresql/microservices/master-info/master-info-service/src/main/java/com/csoftz/master/info/service/interfaces/ICustomerService.java
/*----------------------------------------------------------------------------*/
/* Source File:   ICUSTOMERSERVICE.JAVA                                       */
/* Description:   Interface for Service to retrieve customer information      */
/* Author:        Carlos Adolfo Ortiz Quirós (COQ)                            */
/* Date:          Nov.26/2017                                                 */
/* Last Modified: Dec.06/2017                                                 */
/* Version:       1.1                                                         */
/* Copyright (c), 2017 CSoftZ                                                 */
/*----------------------------------------------------------------------------*/
/*-----------------------------------------------------------------------------
 History
 Nov.26/2017  COQ  File created.
 -----------------------------------------------------------------------------*/
package com.csoftz.master.info.service.interfaces;

import java.util.List;

import com.csoftz.master.info.entity.CustomerEntity;

/**
 * Interface for Service to retrieve customer information
 *
 * @author Carlos Adolfo Ortiz Quirós (COQ)
 * @version 1.1, Dec.06/2017
 * @since 1.8 (JDK), Nov.26/2017
 */
public interface ICustomerService {

	/**
	 * deletes all records
	 */
	void deleteAll();

	/**
	 * Save information
	 * 
	 * @param customer
	 *            Data to save
	 */
	void save(CustomerEntity customer);

	/**
	 * View all the information for all customers
	 * 
	 * @return List of customers
	 */
	List<CustomerEntity> findAll();

	/**
	 * Retrieves information about one customer
	 * 
	 * @param customerId
	 *            The customer Id to set
	 * @param brandcode
	 *            The customer Id to set
	 * @return The customer information.
	 */
	CustomerEntity findOne(int customerId, String brandcode);
}

File: app-code/kotlin/spring-boot-kotlin-postgresql/microservices/orders-info/orders-info-service/src/main/kotlin/com/csoftz/orders/info/service/interfaces/ICustomerService.kt
/*----------------------------------------------------------------------------*/
/* Source File:   ICUSTOMERSERVICE.KT                                         */
/* Description:   Interface for Service to retrieve Customer information      */
/* Author:        Carlos Adolfo Ortiz Quirós (COQ)                            */
/* Date:          Dec.14/2017                                                 */
/* Last Modified: Dec.18/2017                                                 */
/* Version:       1.1                                                         */
/* Copyright (c), 2017 CSoftZ                                                 */
/*----------------------------------------------------------------------------*/
/*-----------------------------------------------------------------------------
 History
 Dec.14/2017  COQ  File created.
 -----------------------------------------------------------------------------*/
package com.csoftz.orders.info.service.interfaces

import com.csoftz.orders.info.entity.CustomerEntity

/**
 * Inteface for Service to retrieve Customer information.
 *
 * @author Carlos Adolfo Ortiz Quirós (COQ)
 * @version 1.1, Dec.18/2017
 * @since 1.8 (JDK), Dec.14/2017
 */
interface ICustomerService {
    /**
     * View all the information for all customers
     *
     *  @return List of customers
     */
    fun findAll(): List<CustomerEntity>
}